File: modulewh/serverwh/src/main/java/com/redsun/server/wh/service/MaterialimportdetailService.java

package com.redsun.server.wh.service;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.redsun.server.wh.model.Materialimportdetail;
import com.redsun.server.wh.model.common.SearchCriteria;

public interface MaterialimportdetailService {

	/**
	 * Saves (create or update) the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 */
	Materialimportdetail save(Materialimportdetail materialimportdetail);

	/**
	 * Create the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 */
	Materialimportdetail create(Materialimportdetail materialimportdetail, Integer idstore)throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateLock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @param idlock
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateUnlock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialimportdetail update(Integer id, Materialimportdetail materialimportdetail) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialimportdetail updateWithLock(Integer id, Materialimportdetail materialimportdetail, Integer idstore)throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialimportdetail updateForDelete(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialimportdetail updateForDeleteWithLock(Integer id, Integer version, Integer idstore)throws JsonParseException, JsonMappingException, IOException;

	
	/**
	 * Deletes the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialimportdetail
	 * @return true if found and deleted, false if not found
	 */
	void delete(Materialimportdetail materialimportdetail);

	/**
	 * Deletes the entity by its Primary Key <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return true if found and deleted, false if not found
	 */
	void deleteById(Integer id);

	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	Materialimportdetail getById(Integer id);

	/**
	 * Loads ALL the entities (use with caution)
	 * @return
	 */
	List<Materialimportdetail> listAll();

	/**
	 * Count all the occurrences
	 * @return
	 */
	long countAll();

	/**
	 * Check exist
	 * @return
	 */
	boolean isExist(Integer id);

	/**
	 * Search by a criteria.
	 * @return
	 */
	List<Materialimportdetail> listWithCritera(SearchCriteria searchCriteria);

	/**
	 * Search by multiple criteria.
	 * @return
	 */
	List<Materialimportdetail> listWithCriteras(List<SearchCriteria> searchCriterias);

	/**
	 * Loads ALL the entities by page (use with caution)
	 * @return
	 */
	Page<Materialimportdetail> listAllByPage(Pageable pageable);

	/**
	 * Search by a criteria by page.
	 * @return
	 */
	Page<Materialimportdetail> listWithCriteraByPage(SearchCriteria searchCriteria, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Materialimportdetail> listWithCriterasByPage(List<SearchCriteria> searchCriterias, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Map<String, Object>> listWithCriteriasByIdmaterialimportAndPage(Integer idmaterialimport, List<SearchCriteria> searchCriterias, Pageable pageable);

	/**
	 * Get quantity by id material import.
	 * @return
	 */
	Map<String, Object> quantityMaterialByIdMaterialimport(Integer idmaterialimport, Integer idmaterial);
	
	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	List<Map<String, Object>> getAllById(Integer id);
	
	Map<String, Object> getByIdForView(Integer id);
	
	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id,price
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	void updatePrice(Integer id, Double price) throws JsonParseException, JsonMappingException, IOException ;
	
	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id,quantity
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	void updateQuantity(Integer id, Integer quantity, Integer idstore) throws JsonParseException, JsonMappingException, IOException ;	
}

File: modulewh/clientwh/src/main/resources/static/js/service/materialimportdetailService.js


/**
 * Service for Materialimportdetail
 **/

define(['require', 'angular'], function (require, angular) {
app.aService(clientwh.prefix + 'materialimportdetailService', function($http, $rootScope) {
	
	// Create.
	this.create = function(materialimportdetail,idstore) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/create/'+idstore;
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
	
	// Update Lock.
	this.updateLock = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateLock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update Unlock.
	this.updateUnlock = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateUnlock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update.
	this.update = function(id, materialimportdetail) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/update/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
	
	// Update With Lock.
	this.updateWithLock = function(id, materialimportdetail, idstore) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateWithLock/' + id + '/'+idstore;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
	
	// Update For Delete.
	this.updateForDelete = function(id, version) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateForDelete/' + id + '/' + version;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update For Delete With Lock.
	this.updateForDeleteWithLock = function(id, version, store) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateForDeleteWithLock/' + id + '/' + version+'/'+store;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Delete.
	this.delete = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/delete/' + id;
		var request = {
				 method: 'DELETE',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getById = function(materialimportdetailId) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/getById/' + materialimportdetailId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getVersionById = function(materialimportdetailId) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/getVerisonById/' + materialimportdetailId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// List for page and filter.
	this.listWithCriteriasByPage = function(criterias, pageNo, pageSize, sorts) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/listWithCriteriasByPage?' + 'page=' + pageNo + '&size=' + pageSize;
		if(typeof(sorts) !== 'undefined' && sorts.length > 0) {
			angular.forEach(sorts, function(sort) {
				serverUrl += '&' + sort;
			});
		}
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: criterias
				}
		return $http(request);
	}
	
	// List for page and filter.
	this.listWithCriteriasByIdmaterialimportAndPage = function(idmaterialimport, criterias, pageNo, pageSize, sorts) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/listWithCriteriasByIdmaterialimportAndPage/' + idmaterialimport + '?page=' + pageNo + '&size=' + pageSize;
		if(typeof(sorts) !== 'undefined' && sorts.length > 0) {
			angular.forEach(sorts, function(sort) {
				serverUrl += '&' + sort;
			});
		}
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: criterias
				}
		return $http(request);
	}
	// Get quantity by material.
	this.getQuantityByMaterial = function (idmaterialimport,idmaterial) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/getQuantityByMaterial/' + idmaterialimport+'/'+idmaterial;
		var request = {
			method: 'GET',
			url: serverUrl
		}
		return $http(request);
	}
			
	// Update With Lock.
	this.updateWithLockForImport = function(id, materialimportdetail) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateWithLockForImport/' + id;
		var request = {
			method: 'PUT',
			url: serverUrl,
			data: materialimportdetail
		}
		return $http(request);
	}
	
	// Get All by Id.
	this.getAllById = function(materialimportdetailId) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/getAllById/' + materialimportdetailId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Create.
	this.createWithSelectDetail = function(materialimportdetail,idstore) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/createWithSelectDetail/'+idstore;
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getByIdForView = function(materialimportdetail) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/getByIdForView/' + materialimportdetail;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update price.
	this.materialimportUpdatePrice = function(materialimportdetail) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updatePrice/';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
	
	// Update quantity.
	this.materialimportUpdateQuantity = function(materialimportdetail,idstore) {
		var serverUrl = clientwh.serverUrl + '/materialimportdetail/updateQuantity/'+idstore;
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialimportdetail
				}
		return $http(request);
	}
});

});
